plugin/flags_sources/cmake_file.py

```diff
@@ -0,0 +1,127 @@
+"""Stores a class that manages flags generation using cmake.
+
+Attributes:
+    log (TYPE): Description
+"""
+from .compilation_db import CompilationDb
+from ..tools import File
+from ..tools import Tools
+from ..tools import SearchScope
+
+from os import path
+
+import subprocess
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class CMakeFile(CompilationDb):
+    """Manages generating a compilation database with cmake.
+
+    Attributes:
+        cache (dict): Cache of all parsed cmake files to date.
+        path_for_file (dict): A path to a database for every source file path.
+    """
+    _FILE_NAME = 'CMakeLists.txt'
+    _CMAKE_MASK = 'cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON "{path}"'
+
+    cache = {}
+    path_for_file = {}
+
+    def __init__(self, include_prefixes, search_scope, prefix_paths):
+        """Initialize a cmake-based flag storage.
+
+        Args:
+            include_prefixes (str[]): A List of valid include prefixes.
+            search_scope (SearchScope): Where to search for a CMakeLists.txt.
+            prefix_paths (str[]): A list of paths to append to
+                CMAKE_PREFIX_PATH before invoking cmake.
+        """
+        super(CMakeFile, self).__init__(include_prefixes, search_scope)
```

Comment:
nit: If you don't care about Python/Sublime 2 (I don't think you do) then you can just use `super().__init__`